sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -393,36 +420,268 @@ def tune(
+                )
+                from kubeflow.storage_initializer.s3 import S3DatasetParams
+                from kubeflow.training.constants.constants import STORAGE_INITIALIZER
+                from kubeflow.training.constants.constants import (
+                    STORAGE_INITIALIZER_IMAGE,
+                )
+                from kubeflow.training.constants.constants import (
+                    STORAGE_INITIALIZER_VOLUME_MOUNT,
+                )
+                from kubeflow.training.constants.constants import (
+                    TRAINER_TRANSFORMER_IMAGE,
+                )
+            except ImportError:
+                raise ImportError(
+                    "Tune API dependencies not installed. "
+                    + "Run: pip install -U 'kubeflow-katib[huggingface]' "
+                )
+
+            # Create PVC for the Storage Initializer.
+            # TODO (helenxie-bit): PVC Creation should be part of Katib Controller.
+            try:
+                self.core_api.create_namespaced_persistent_volume_claim(
+                    namespace=namespace,
+                    body=utils.get_pvc_spec(
+                        pvc_name=name,
+                        namespace=namespace,
+                        storage_config=storage_config,
+                    ),
+                )
+            except Exception as e:
+                pvc_list = self.core_api.list_namespaced_persistent_volume_claim(
+                    namespace
+                )
+                # Check if the PVC with the specified name exists.
+                for pvc in pvc_list.items:
+                    if pvc.metadata.name == name:
+                        print(
+                            f"PVC '{name}' already exists in namespace " f"{namespace}."
+                        )
+                        break
+                else:
+                    raise RuntimeError(f"failed to create PVC. Error: {e}")
+
+            if isinstance(model_provider_parameters, HuggingFaceModelParams):
+                mp = "hf"
+            else:
+                raise ValueError(
+                    "Model provider parameters must be an instance of HuggingFaceModelParams."
+                )
+
+            if isinstance(dataset_provider_parameters, S3DatasetParams):
+                dp = "s3"
+            elif isinstance(dataset_provider_parameters, HuggingFaceDatasetParams):
+                dp = "hf"
+            else:
+                raise ValueError(
+                    "Dataset provider parameters must be an instance of S3DatasetParams "
+                    "or HuggingFaceDatasetParams."
+                )
+
+            # Iterate over input parameters.
+            experiment_params = []
+            trial_params = []
+
+            training_args = trainer_parameters.training_parameters
+            for (
+                p_name,
+                p_value,
+            ) in trainer_parameters.training_parameters.to_dict().items():
```

Comment:
Given that loop for lora_config (e.g. L601) and training_parameters are almost identical, can you create helper function in Katib SDK util to reduce code duplication please ?